C++ parser tests almost all pass

cnewman opened the pull request:
Since this is a large chunk on its own, figured I'd submit for now.

Since the change was partially automated, some of the names aren't so great (as @patrickfrancis has pointed out).  I can tackle this after the rest of it is done since I can pattern match variable names.

I wasn't sure what to do with OP/LIT since I think it was stated that backwards compatibility was wanted? So if that's the case, I can put OP back where it was but I think abstract code parser was pretty tightly coupled to it. So backwards compatibility might be weird to try and do.

I've also just noticed a stray comment I didn't remove:
//NOTE: When the declaration is using f = bla this won't work because there is no decl on the inside. Fix.

Java/C# are going okay; some of their issues were fixed as I fixed C++.  Will push those as I complete them.

cnewman committed:
A lot of modifications to parser code to handle new format. LIT and OP have been removed. POS has had a tag added to its namespace; the position tag.

cnewman committed:
Changes to abstract code parser and several unit tests. Mostly removing some of the bad parts of the auto transformation I wrote to convert the unit tests (there're still more to remove) and a lot of changing things in the parser or in DSes the parser uses so that it expects the new version of srcML. Most CPP tests passing at this point.

cnewman committed:
asdsadas

cnewman committed:
Revert "asdsadas"

This reverts commit 9336b0934a71168f025444da5baa95a22a0b6231.

cnewman committed:
C++ parser tests mostly pass. 4 of them don't-- will probably skip for now and finish c#/java then come back

cnewman committed:
Transformation introduced silly syntax. Not really a bug but removing anyway.

vinayaugustine commented:
IMO, we can break backwards compatibility if srcML does. If someone wants to use the OP/LIT namespaces they can use older versions of SrcML.NET.

cnewman committed:
C# tests all pass with the exception of 4. Will look at those tomorrow.

cnewman committed:
Only 4 tests don't pass in each language-specific parser. Haven't looked at other tests yet.

cnewman committed:
All javaparser.cs tests pass

cnewman committed:
Only 3 tests not passing between java/c++/c# parser.cs files. Working on the codeparser.cs tests still.

cnewman committed:
Small commit before larger changes

cnewman committed:
More test updates.

cnewman committed:
Converted the rest of the tests to libsrcmlrunner

cnewman committed:
Changed library to not assume .cpp. Don't know why it was in the first place.

Added a note on the mapping tests

updated other file-read tests to use lib.

cnewman commented:
Currently 17 tests fail but they're mostly reading/writing srcML files using the old runner. Need to change the way it's reading in/generating srcml/then reading out files and things like that.

The parser tests are working except for a handful-- 3 or 4? maybe 5. 

I went ahead and pushed this into the pull for now since next week is my last. Tomorrow I'll look over this to make sure everything is in order/make changes where needed. It's big enough that there might still be a bit of mess in places.

cnewman committed:
Changes to how files are named by the library broke some tests.

Also, I apparently didn't commit the xsl script I used to test the xslt portion of the library so I had to recreate it.

cnewman committed:
These tests weren't in srcml.data but I converted them anyway.

cnewman committed:
I started swapping tests in this file but then realized it's the generator tests for the other srcml generator. Not sure if this should happen or not.
